cmd/cluster-agent/app/app.go

```diff
@@ -0,0 +1,166 @@
+package app
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	_ "expvar" // Blank import used because this isn't directly used in this file
+
+	"github.com/DataDog/datadog-agent/cmd/agent/common"
+	"github.com/DataDog/datadog-agent/cmd/cluster-agent/api"
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/clusteragent"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/forwarder"
+	"github.com/DataDog/datadog-agent/pkg/serializer"
+	"github.com/DataDog/datadog-agent/pkg/util"
+	"github.com/DataDog/datadog-agent/pkg/version"
+	log "github.com/cihub/seelog"
+	"github.com/spf13/cobra"
+)
+
+// FIXME: move SetupAutoConfig and StartAutoConfig in their own package so we don't import cmd/agent
+var (
+	ClusterAgentCmd = &cobra.Command{
+		Use:   "cluster-agent [command]",
+		Short: "Datadog Cluster Agent at your service.",
+		Long: `
+Datadog Cluster Agent takes care of running checks that need run only once per cluster.
+It also exposes an API for other Datadog agents that provides them with cluster-level
+metadata for their metrics.`,
+	}
+
+	startCmd = &cobra.Command{
+		Use:   "start",
+		Short: "Start the Cluster Agent",
+		Long:  `Runs Datadog Cluster agent in the foreground`,
+		RunE:  start,
+	}
+
+	versionCmd = &cobra.Command{
+		Use:   "version",
+		Short: "Print the version number",
+		Long:  ``,
+		Run: func(cmd *cobra.Command, args []string) {
+			av, _ := version.New(version.AgentVersion, version.Commit)
+			fmt.Println(fmt.Sprintf("Cluster Agent from Agent %s - Codename: %s - Commit: %s - Serialization version: %s", av.GetNumber(), av.Meta, av.Commit, serializer.AgentPayloadVersion))
+		},
+	}
+
+	confPath    string
+	flagNoColor bool
+)
+
+func init() {
+	// attach the command to the root
+	ClusterAgentCmd.AddCommand(startCmd)
+	ClusterAgentCmd.AddCommand(versionCmd)
+
+	ClusterAgentCmd.PersistentFlags().StringVarP(&confPath, "cfgpath", "c", "", "path to directory containing datadog-cluster.yaml")
+	ClusterAgentCmd.PersistentFlags().BoolVarP(&flagNoColor, "no-color", "n", false, "disable color output")
+}
+
+func start(cmd *cobra.Command, args []string) error {
+	configFound := false
+
+	// a path to the folder containing the config file was passed
+	if len(confPath) != 0 {
+		// we'll search for a config file named `datadog-cluster.yaml`
```

Comment:
I'd rather use a `datadog.yaml` file for consistency, WDYT?

Reply:
I am afraid users might confuse both files (when mounting it for instance)

Reply:
fair enough